lib/semantics/resolve-names.cc

```diff
@@ -875,7 +875,6 @@ class DeclarationVisitor : public ArraySpecVisitor,
 class ConstructVisitor : public DeclarationVisitor {
 public:
   bool Pre(const parser::ConcurrentHeader &);
```

Comment:
Are there any plans to replace these checker-specific Pre/Post calls with semantics-general Enter/Leave functions?  I believe that would be more uniform, and avoid this extra set of calls specific only to do checking.  If so, that might best be done sooner rather than later, so that any new additions use the new framework.

Reply:
@vdonaldson , you might be right.  In particular, if you look at the various semantic checking files in .../lib/semantics, you'll see a fair amount of variation.  For me personally, I plan to keep working on individual DO statement constraints.  As I'm working on the code, I'm also learning a lot more about it.  So I feel like I'll be in a much better position to decide whether and how to refactor the code after finishing my current goal of completing semantic checking for DO loops.